### PR TITLE
Fix linear argument

### PR DESCRIPTION
**Pull request by Yancey1989**
related  issue #113 

This PR fixed the problem that the initialized weight of FC layers are different between C++ and Go. And the MNIST example gets a  closer loss to the C++ version, we should continue to find out the reason to let  C++ and Go got the same loss.

Gotorch MNIST example  loss:

``` text
2020/08/15 23:45:41 Epoch: 0, Loss: 0.1583
2020/08/15 23:45:46 Epoch: 1, Loss: 0.0852
```

### CODE REVIEW

File: mnist_test.go [hunk 1]
```diff
@@ -18,9 +18,9 @@ type MLPMNISTNet struct {
 
 func NewMNISTNet() *MLPMNISTNet {
 	r := &MLPMNISTNet{
-		FC1: nn.Linear(28*28, 512, false),
-		FC2: nn.Linear(512, 512, false),
-		FC3: nn.Linear(512, 10, false)}
+		FC1: nn.Linear(28*28, 512, true),
```
Comment (wangkuiyi) [inline mnist_test.go:21]: Can you please explain why would this work on random seed initialization?

File: nn/linear.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ func Linear(in, out int64, bias bool) *LinearModule {
 	}
 	l.Weight = torch.Empty([]int64{out, in}, true)
 	if bias {
-		l.Bias = torch.Empty([]int64{out, 1}, true)
+		l.Bias = torch.Empty([]int64{out}, true)
```
Comment (wangkuiyi) [inline nn/linear.go:29]: This the change on this line affect random initialization?

File: mnist_test.go [hunk 3]
```diff
@@ -35,7 +35,7 @@ func (n *MLPMNISTNet) Forward(x torch.Tensor) torch.Tensor {
 	return x.LogSoftmax(1)
 }
 
-func ExampleTrainMNIST() {
+func ExampleTrainMNISTMLP() {
```
Comment (wangkuiyi) [inline mnist_test.go:38]: ExampleTrainMLPUsingMNIST ?

File: mnist_test.go (hunk 1 above)
Comment (Yancey1989) [reply mnist_test.go:21]: The `Linear` enables the `bias` by default in C++, I think it's just a bug in the previous implementation in Go.
c.f. https://github.com/pytorch/pytorch/blob/v1.6.0/torch/csrc/api/include/torch/nn/options/linear.h#L25

File: nn/linear.go (hunk 2 above)
Comment (Yancey1989) [reply nn/linear.go:29]: No, the bias shape should be `{out}`, it's a  bug, c.f. https://github.com/pytorch/pytorch/blob/v1.6.0/torch/csrc/api/src/nn/modules/linear.cpp#L36
